src/service/search/cache/cacher.rs

```diff
@@ -519,21 +521,64 @@ pub fn calculate_deltas(
 }
 
 pub async fn cache_results_to_disk(
+    trace_id: &str,
     file_path: &str,
     file_name: &str,
     data: String,
+    clear_cache: bool,
+    clean_start_ts: Option<i64>,
+    clean_end_ts: Option<i64>,
```

Comment:
as we discussed, we can clean cache first before run the search. then we don't need additional logic. just run `cache::search`.